alignak_backend/livesynthesis.py

```diff
@@ -214,51 +264,79 @@ def on_inserted_service(items):
                 ls.recalculate()
             else:
                 typecheck = 'services'
-                data = {"$inc": {"%s_%s_%s" % (typecheck, item['ls_state'].lower(),
-                                               item['ls_state_type'].lower()): 1,
-                                 "%s_total" % (typecheck): 1}}
+                if not item['active_checks_enabled'] and not item['passive_checks_enabled']:
+                    data = {"$inc": {"%s_not_monitored" % typecheck: 1,
+                                     "%s_total" % typecheck: 1}}
+                else:
+                    data = {"$inc": {"%s_%s_%s" % (typecheck, item['ls_state'].lower(),
+                                                   item['ls_state_type'].lower()): 1,
+                                     "%s_total" % typecheck: 1}}
+                if 'ALIGNAK_BACKEND_PRINT' in os.environ:
+                    print("LS - inserted service %s: %s..." % (item['name'], data))
                 current_app.data.driver.db.livesynthesis.update({'_id': live_current['_id']}, data)
 
     @staticmethod
     def on_updated_host(updated, original):
         """
             What to do when an host live state is updated ...
+
+            If the host monitored state is changing, recalculate the live synthesis, else
+            simply update the livesynthesis
         """
         if original['_is_template']:
             return
 
+        # If the host is not monitored and we do not change its monitoring state
+        if not original['active_checks_enabled'] and not original['passive_checks_enabled'] \
+                and 'active_checks_enabled' not in updated \
+                and 'passive_checks_enabled' not in updated:
+            return
+
         minus, plus = Livesynthesis.livesynthesis_to_update('hosts', updated, original)
         if minus:
             livesynthesis_db = current_app.data.driver.db['livesynthesis']
             live_current = livesynthesis_db.find_one({'_realm': original['_realm']})
-            if live_current is None:
+            if live_current is None or 'not_monitored' in minus or 'not_monitored' in plus:
```

Comment:
If not not have `not_monitored` in `minus` or `plus` (the most cases), you will recalculate, so it's not good.
Need add in tests if the `_id` of the livesynthesis in the same when update a host / service to be sure it's not recalculate

Reply:
No.
If `live_current` does not exist, 
Or `not_monitored` in minus ...
Or `not_monitored` in plus ...
then recalculate ... thus the recalculation happens only if the livesynthsis does not exist or when the monitored status is changing! and this does not happen very often ... only when the `active_checks_enabled` or `passive_checks_enabled` changed. ..

Reply:
Ok so, why need recalculate when one of both changed? I mean, you have minus / plus so not need recalculate.

Reply:
Ok we keep that